src/script_movement.c

```diff
@@ -0,0 +1,226 @@
+    taskId = GetMoveObjectsTaskId();
+    moveScrId = GetMovementScriptIdFromObjectEventId(taskId, objEventId);
+    if (moveScrId == OBJECT_EVENTS_COUNT)
+        return TRUE;
+    return IsMovementScriptFinished(taskId, moveScrId);
+}
+
+void ScriptMovement_UnfreezeObjectEvents(void)
+{
+    u8 taskId = GetMoveObjectsTaskId();
+    if (taskId != TAIL_SENTINEL)
+    {
+        ScriptMovement_UnfreezeActiveObjects(taskId);
+        DestroyTask(taskId);
+    }
+}
+
+void ScriptMovement_StartMoveObjects(u8 priority)
+{
+    u8 i;
+    u8 taskId = CreateTask(ScriptMovement_MoveObjects, priority);
+    for (i = 1; i < NUM_TASK_DATA; i++)
+    {
+        gTasks[taskId].data[i] = -1;
+    }
+}
+
+u8 GetMoveObjectsTaskId(void)
+{
+    return FindTaskIdByFunc(ScriptMovement_MoveObjects);
+}
+
+bool8 ScriptMovement_TryAddNewMovement(u8 taskId, u8 objEventId, const u8 *movementScript)
+{
+    u8 moveScrId;
+
+    moveScrId = GetMovementScriptIdFromObjectEventId(taskId, objEventId);
+    if (moveScrId != OBJECT_EVENTS_COUNT)
+    {
+        if (IsMovementScriptFinished(taskId, moveScrId) == FALSE)
+        {
+            return TRUE;
+        }
+        else
+        {
+            ScriptMovement_AddNewMovement(taskId, moveScrId, objEventId, movementScript);
+            return FALSE;
+        }
+    }
+    moveScrId = GetMovementScriptIdFromObjectEventId(taskId, OBJ_EVENT_ID_PLAYER);
+    if (moveScrId == OBJECT_EVENTS_COUNT)
+    {
+        return TRUE;
+    }
+    else
+    {
+        ScriptMovement_AddNewMovement(taskId, moveScrId, objEventId, movementScript);
+        return FALSE;
+    }
+}
+
+u8 GetMovementScriptIdFromObjectEventId(u8 taskId, u8 objEventId)
+{
+    u8 i;
+    u8 *moveScriptId = (u8 *)&gTasks[taskId].data[1];
+    for (i = 0; i < OBJECT_EVENTS_COUNT; i++, moveScriptId++)
+    {
+        if (*moveScriptId == objEventId)
+            return i;
+    }
+    return OBJECT_EVENTS_COUNT;
+}
+
+void LoadObjectEventIdPtrFromMovementScript(u8 taskId, u8 moveScrId, u8 **pObjEventId)
+{
+    u8 i;
+
+    *pObjEventId = (u8 *)&gTasks[taskId].data[1];
+    for (i = 0; i < moveScrId; i++, (*pObjEventId)++)
+        ;
+}
+
+void SetObjectEventIdAtMovementScript(u8 taskId, u8 moveScrId, u8 objEventId)
+{
+    u8 *ptr;
+
+    LoadObjectEventIdPtrFromMovementScript(taskId, moveScrId, &ptr);
+    *ptr = objEventId;
+}
+
+void LoadObjectEventIdFromMovementScript(u8 taskId, u8 moveScrId, u8 *objEventId)
+{
+    u8 *ptr;
+
+    LoadObjectEventIdPtrFromMovementScript(taskId, moveScrId, &ptr);
+    *objEventId = *ptr;
+}
+
+
+static void ClearMovementScriptFinished(u8 taskId, u8 moveScrId)
+{
+    u16 mask = ~gBitTable[moveScrId];
+
+    gTasks[taskId].data[0] &= mask;
+}
+
+static void SetMovementScriptFinished(u8 taskId, u8 moveScrId)
+{
+    gTasks[taskId].data[0] |= gBitTable[moveScrId];
+}
+
+static bool8 IsMovementScriptFinished(u8 taskId, u8 moveScrId)
+{
+    u16 moveScriptFinished = (u16)gTasks[taskId].data[0] & gBitTable[moveScrId];
```

Comment:
`bool16`

Reply:
`bool` implies a value of 0 or 1, which may not be the case here.

Reply:
Oh fuck I'm blind